### DISCUSSION TITLE
How to tell that fetcher.submit is finished?

### DISCUSSION THREAD

**Discussion by tomer-yechiel** [remix-run/remix#4629] [Proposals]
Hi,
I want to reload the page without browser navigation.
so I am using useFetcher.

I want to use the API  fetcher.submit but it returns void and not Promise so I can't tell when the operation is done.
is there a way to know that beside using the state field like fetcher.state?

if there isn't please consider to change the API

thanks!

**Comment by sergiodxa** [comment]
The only way is to check the `fetcher.state`
